pydicom/sr/value_types.py

```diff
@@ -0,0 +1,889 @@
+        self.ReferencedSOPSequence = [item]
+
+
+class ScoordContentItem(ContentItem):
+
+    """DICOM SR document content item for value type SCOORD.
+
+    Note
+    ----
+    Spatial coordinates are defined in image space and have pixel units.
+
+    """
+
+    def __init__(self, name, graphic_type, graphic_data,
+                 pixel_origin_interpretation, fiducial_uid=None,
+                 relationship_type=None):
+        """
+        Parameters
+        ----------
+        name: Union[pydicom.sr.coding.CodedConcept, pydicom.sr.coding.Code]
+            concept name
+        graphic_type: Union[pydicom.sr.value_types.GraphicTypes, str]
+            name of the graphic type
+        graphic_data: Union[List[List[Union[int, float]]], List[Union[int, float]]]
+            ordered (column, row) coordinate pairs
+        pixel_origin_interpretation: Union[pydicom.sr.value_types.PixelOriginInterpretations, str]
+            whether pixel coordinates specified by `graphic_data` are defined
+            relative to the total pixel matrix
+            (``pydicom.sr.value_types.PixelOriginInterpretations.VOLUME``) or
+            relative to an individual frame
+            (``pydicom.sr.value_types.PixelOriginInterpretations.FRAME``)
+        fiducial_uid: Union[pydicom.uid.UID, str, None], optional
+            unique identifier for the content item
+        relationship_type: Union[pydicom.sr.value_types.RelationshipTypes, str, None], optional
+            type of relationship with parent content item
+
+        """  # noqa
+        super(ScoordContentItem, self).__init__(
+            ValueTypes.SCOORD, name, relationship_type
+        )
+        graphic_type = GraphicTypes(graphic_type)
+        pixel_origin_interpretation = PixelOriginInterpretations(
+            pixel_origin_interpretation
+        )
+        self.GraphicType = graphic_type.value
+
+        def is_point(coordinate):
+            try:
+                return all([
+                    isinstance(coordinate, (list, tuple, )),
+                    len(coordinate) == 2,
+                    all([isinstance(c, (int, float, )) for c in coordinate]),
+                ])
+            except IndexError:
+                return False
+
+        if graphic_type == GraphicTypes.POINT:
+            if not is_point(graphic_data):
+                raise ValueError(
+                    'Graphic data of a scoord of graphic type "POINT" '
+                    'must be a single (column row) pair in two-dimensional '
+                    'image coordinate space.'
+                )
+            # Flatten list of coordinate pairs
+            self.GraphicData = [
+                float(coordinate) for coordinate in graphic_data
+            ]
+        else:
+            # TODO: This may be expensive for large lists.
+            are_all_points = all([
+                is_point(coordinates) for coordinates in graphic_data
+            ])
+            if graphic_type == GraphicTypes.CIRCLE:
```

Comment:
Another spot where list brackets should be removed -- especially as per comment, if large lists then full list composed in memory)